test/tests/functional/pbs_multi_sched.py

```diff
@@ -1637,3 +1637,103 @@ def test_dedicated_time_multisched(self):
         self.server.expect(JOB, {ATTR_comment: "Not Running: " + msg}, id=jid2)
         self.scheds['sc2'].log_match(jid2 + ";Dedicated Time",
                                      starttime=self.server.ctime)
+
+    def test_auto_sched_off_due_to_fds_limit(self):
+        """
+        Test to make sure scheduling should be turned off automatically
+        when number of open files per process are exhausted
+        """
+
+        # get the number of open files per process
+        ret = self.du.run_cmd(
+            self.server.hostname, [
+                'ulimit', '-n'], as_script=True, logerr=False)
+        self.assertEqual(ret['rc'], 0)
+        open_files_limit = ret['out'][0]
+
+        # set the number of open files per process to 10
+        ret = self.du.run_cmd(
+            self.server.hostname, [
+                'ulimit', '-n', '10'], as_script=True, logerr=False)
+        self.assertEqual(ret['rc'], 0)
+
+        self.setup_sc3()
+
+        for _ in range(0, 100):
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'scheduling': 'True'}, id="sc3", expect=True)
```

Comment:
This probably won't start 100 scheduling cycles.  When multiple events that would start a cycle during a cycle, we only keep one and start a cycle after the current one finishes.  This means you could set scheduling=true more than once during a cycle and it just means you'll start one cycle afterwards.

Reply:
Thanks for catching this Bhroam. I did not notice it earlier. Now I am making scheduler_iteration=1 second and waiting for more than 10 seconds for making sure that at least 10 scheduling cycles will be completed and it also exhausts the set open files per process which is 10 in this case.

Earlier also for some reason when I run this test case it always used to elapse 10 open files process and hence the test was failing before the fix.